rom/dev/src/flow/warm_reset.rs

```diff
@@ -27,8 +28,15 @@ impl WarmResetFlow {
     /// * `env` - ROM Environment
     #[inline(never)]
     #[cfg_attr(not(feature = "no-cfi"), cfi_impl_fn)]
-    pub fn run(_env: &mut RomEnv) -> CaliptraResult<Option<FirmwareHandoffTable>> {
+    pub fn run(env: &mut RomEnv) -> CaliptraResult<Option<FirmwareHandoffTable>> {
         cprintln!("[warm-reset] ++");
+
+        // Check if previous Cold-Reset was successful.
+        if env.data_vault.rom_cold_boot_status() != ColdResetComplete.into() {
```

Comment:
Can we add some CFI defenses here? Read the register a second time and assert if it's different?

Reply:
Added.